Dockerfile

```diff
@@ -19,10 +19,13 @@ ARG VERSION
 
 WORKDIR /sigs.k8s.io/external-dns
 
-COPY . .
+COPY go.mod .
+COPY go.sum .
 RUN go mod vendor && \
-    make test && \
-    make build
+    go mod download
+
+COPY . .
```

Comment:
why not running COPY . . before instead of COPY go.mod and COPY go.sum and then again COPY . .?

Reply:
It's a mico-optimization as part of docker build. If the shasum hasn't changed it should use a previous intermediate layer

Reply:
I'm sorry for my naive question but are you building the Docker image always manually? I'm assuming here you would use the image we provide for external-dns. Wouldn't it not only matter for building the image, which is something you shouldn't do at a regular basis?

Reply:
This speeds up container image builds for local development (i.e. developers contributing to external-dns)

Reply:
Exactly, it also depends on how someone's CI system works. If they pull down the previous version prior to build you can also get faster builds from that. But this is mostly geared at the developer experience.